src/coreclr/jit/lower.cpp

```diff
@@ -4148,10 +4148,10 @@ GenTree* Lowering::OptimizeConstCompare(GenTree* cmp)
         }
     }
 
-    // Optimize EQ/NE(op_that_sets_zf, 0) into op_that_sets_zf with GTF_SET_FLAGS + SETCC.
+    // Optimize EQ/NE/GT/GE/LT/LE(op_that_sets_zf, 0) into op_that_sets_zf with GTF_SET_FLAGS + SETCC.
     LIR::Use use;
-    if (cmp->OperIs(GT_EQ, GT_NE) && op2->IsIntegralConst(0) && op1->SupportsSettingZeroFlag() &&
-        BlockRange().TryGetUse(cmp, &use))
+    if (cmp->OperIs(GT_EQ, GT_NE, GT_GT, GT_GE, GT_LT, GT_LE) && op2->IsIntegralConst(0) &&
+        op1->SupportsSettingFlags() && BlockRange().TryGetUse(cmp, &use))
```

Comment:
should this be arm64 specific change? seems there are some test failures on x64 as well.

Reply:
Why make this arm64 specific? We should figure out why this would not work for x64 and what would be different between x64 and arm64 in those cases before we just disable this.

Reply:
I wanted to have this focused change for arm64. We should have separate PR if we want to enable it for x64.

Reply:
That's ok with me -- in that case I think we should introduce a new `SupportsSettingFlags` that is returns false on all platforms but arm64, and then make the check something like
```
(cmp->OperIs(GT_EQ, GT_NE) && op1->SupportsSettingZeroFlag()) || (cmp->OperIs(GT_GT, GT_GE, GT_LT, GT_LE) && op1->SupportsSettingResultFlags())
```
(The rename to `SupportsSettingFlags` does not look right for xarch, since that function currently queries `emitter::DoesWriteZeroFlag` for hardware intrinsics. That might be the cause of the failures.)